### PR TITLE
feat:Added bookmarks and explore feed

### PR DESCRIPTION
**Pull request by Hardiegogo**
- Added bookmarks functionality
- Added explore feed

Screenshots:
image
image

### CODE REVIEW

File: src/components/Post.jsx [hunk 1]
```diff
@@ -44,6 +49,13 @@ const Post=({post})=> {
         dispatch(setSelectedPost(post))
         dispatch(postModalOn())
     }
+    const bookmarkClickHandler=()=>{
+        if(!isBookmarked){
+            dispatch(addToBookmarks(_id))
+        }else{
+            dispatch(deleteFromBookmarks(_id))
+        }setIsBookmarked(!isBookmarked)
```
Comment (AmishMishra11) [inline src/components/Post.jsx:57]: You can use ternary operator here

**Review by AmishMishra11 — APPROVED** [review]
Liked the use of compose function
Ui looks good

File: src/components/PostsList.jsx [hunk 2]
```diff
@@ -1,39 +1,58 @@
 import React from "react";
 import { useState } from "react";
```
Comment (dev-enforced) [inline src/components/PostsList.jsx:2]: Can combine both in 1 line.

File: src/utils/post-utils/post-services.js [hunk 3]
```diff
@@ -108,4 +108,51 @@ export const editPost=createAsyncThunk('posts/editPost',async(postData)=>{
...
+        const res=await axios({
+            method:"POST",
+            url:`/api/users/remove-bookmark/${postId}`,
+            headers:{authorization:encodedToken},
+        })
+        if(res.status===200){
+            return res.data.bookmarks
+        }
+    } catch (error) {
+        console.log("error occured",error)
```
Comment (dev-enforced) [inline src/utils/post-utils/post-services.js:140]: Can use console.error to get error messages on the console helps in tracking and differentiating down further.

File: src/utils/post-utils/filterByFollowing.js [hunk 4]
```diff
@@ -1,15 +1,37 @@
...
+    return { filteredPosts, sortBy, feedType, currentUser, selectedUser };
+  } else
+    return {
+      filteredPosts: posts,
+      sortBy,
+      feedType,
+      currentUser,
+      selectedUser,
+    };
+};
```
Comment (dev-enforced) [inline src/utils/post-utils/filterByFollowing.js:37]: Liked the logic handling in this part and the way you used it in the postLists.jsx file along with other sub functions. Good work @Hardiegogo

**Review by dev-enforced — APPROVED** [review]

File: src/utils/post-utils/filterByFollowing.js (hunk 4 above)
Comment (Hardiegogo) [reply src/utils/post-utils/filterByFollowing.js:37]: Thank you Roshan

File: src/components/PostsList.jsx (hunk 2 above)
Comment (Hardiegogo) [reply src/components/PostsList.jsx:2]: Got it

File: src/components/Post.jsx (hunk 1 above)
Comment (Hardiegogo) [reply src/components/Post.jsx:57]: Okay

**Pull request merged**